### PR TITLE
Store interim counts in Redis

### PR DESCRIPTION
**Pull request by DJAndries**
Resolves https://github.com/brave/go-sync/issues/256

### CODE REVIEW

File: datastore/item_count.go [hunk 1]
```diff
@@ -49,7 +51,7 @@ func (counts *ClientItemCounts) SumHistoryCounts() int {
 
 func (dynamo *Dynamo) initRealCountsAndUpdateHistoryCounts(counts *ClientItemCounts) error {
 	now := time.Now().Unix()
-	if counts.LastPeriodChangeTime == 0 {
+	if counts.Version < currentCountVersion {
```
Comment (bsclifton) [inline datastore/item_count.go:54]: If this field is new (`.Version`), does it default to 0 / able to do comparison? Expecting it should evaluate as true - then it's setting to `currentCountVersion` (2). Then next time, you might manually increase `currentCountVersion` to 3 (if something changes about how we do count)?

File: cache/redis.go [hunk 2]
```diff
@@ -85,8 +104,25 @@ func (r *redisClusterClient) Set(ctx context.Context, key string, val string, tt
 	return r.client.Set(ctx, key, val, ttl).Err()
 }
 
-func (r *redisClusterClient) Get(ctx context.Context, key string) (string, error) {
-	val, err := r.client.Get(ctx, key).Result()
+func (r *redisClusterClient) Incr(ctx context.Context, key string, subtract bool) (int, error) {
```
Comment (bsclifton) [inline cache/redis.go:107]: I'm not familiar enough with Go to know about this - but logic is the same as above in redisSimpleClient. Is there a polymorphic way to use the same code? (ex: can redisClusterClient inherit and not override Incr/Get)

**Review by bsclifton — APPROVED** [review]
I'm not super familiar with nuances of Go and much of the code-base in this repo - but the logic added LGTM for solving the issue linked.

File: cache/redis.go (hunk 2 above)
Comment (DJAndries) [reply cache/redis.go:107]: perhaps there is a way, but might be best to follow up once things are stable

**Review by AlexeyBarabash — APPROVED** [review]
++

**Pull request merged**